### PR TITLE
Added new tests in test_clique.py

### LINKED ISSUE #6141: Improve test coverage in clique.py
Currently `clique.py` in the Algorithms section doesn't have full coverage. I want to improve that.

### PR DESCRIPTION
**Pull request by paulitapb**
Close #6141 
Added some tests to improve coverage to 100%

### CODE REVIEW

File: networkx/algorithms/tests/test_clique.py [hunk 1]
```diff
@@ -233,6 +233,17 @@ def test_directed(self):
...
 
+    def test_find_cliques_trivial(self):
+        G = nx.Graph()
+        assert sorted(nx.find_cliques(G)) == []
+        assert sorted(nx.find_cliques_recursive(G)) == []
+
+    def test_make_max_clique_graph_2(self):
+        "Test the create_using parameter"
+        G = nx.Graph()
+        G.add_edges_from([(1, 2), (3, 1), (4, 1), (5, 6)])
```
Comment (rossbar) [inline networkx/algorithms/tests/test_clique.py:244]: Just a nit, but might as well simplify this and put it all on one line!
```suggestion
        G = nx.Graph([(1, 2), (3, 1), (4, 1), (5, 6)])
```

File: networkx/algorithms/tests/test_clique.py [hunk 2]
```diff
@@ -233,6 +233,17 @@ def test_directed(self):
...
+        assert sorted(nx.make_max_clique_graph(G, nx.Graph)) == [0, 1, 2, 3]
```
Comment (rossbar) [inline networkx/algorithms/tests/test_clique.py:245]: This check isn't very specific - it basically just checks that there are 4 nodes in the max_clique_graph, but doesn't test anything else about it. What do we expect the output graph of `max_clique_graph` to look like in this case?

Comment (paulitapb) [reply networkx/algorithms/tests/test_clique.py:243]: True, I will modify it so we can check that the edges between the cliques are ok too. With this modification, we know that the cliques are found and the edges between the cliques that share nodes are added. In this case, we have 4 cliques and 3 of them share nodes so these should form a connected component in the maximal clique graph.

**Review by MridulS — APPROVED** [review]

**Review by rossbar — APPROVED** [review]
LGTM thanks @paulitapb !

The CI :x: 's are unrelated, so in this goes!

**Pull request merged**